frontend/apps/reader/modules/readersearch.lua

```diff
@@ -303,6 +303,19 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
                     vsync = true,
                     callback = search(self.searchNext, text, 1),
                 },
+                {
+                    icon = "appbar.search",
+                    vsync = true,
+                    callback = function()
+                        self.ui.highlight:clear()
+                        UIManager:setDirty(self.dialog, "ui")
+                        UIManager:nextTick(function()
+                            UIManager:close(self.search_dialog)
+                        end)
+                        self.last_search_text = text
```

Comment:
Is line 315 really necessary?

Reply:
Yes, when calling Search from the highlight popup menu.